### PR TITLE
find nested vmware clusters

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -775,7 +775,8 @@ def find_cluster(self, name, compute_resource):
 
         # workaround for https://projects.theforeman.org/issues/31874
         if compute_resource['provider'].lower() == 'vmware':
-            cluster['_api_identifier'] = cluster['name']
+            path_or_name = cluster.get('full_path', cluster['name'])
+            cluster['_api_identifier'] = six.moves.urllib.parse.quote(path_or_name, safe='')
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py]: the double quoting (here and in apypie) is required for https://projects.theforeman.org/issues/35438

**Comment by sayan3296** [comment]
I confirm that the changes are working great in both Nested and Non-Nested cluster situations.

File: plugins/module_utils/_apypie.py [hunk 2]
```diff
@@ -901,7 +906,7 @@ def path_with_params(self, params=None):
         if params is not None:
             for param in self.params_in_path:
                 if param in params:
-                    result = result.replace(':{}'.format(param), str(params[param]))
+                    result = result.replace(':{}'.format(param), quote(str(params[param]), safe=''))
```
Comment (evgeni) [inline plugins/module_utils/_apypie.py:909]: this is vendored from https://github.com/Apipie/apypie/pull/139

**Comment by evgeni** [comment]
this has been tested live, I am going to merge it.

**Pull request merged**